File: packages/android/src/main/java/com/kozmos/components/Rating/Rating.figma.kt
package com.kozmos.components.rating

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=512-39747")
class KozmosRatingConnect {
    @FigmaProperty(FigmaType.Enum, "Value")
    val value: Int = Figma.mapping(
        "0" to 0,
        "1" to 1,
        "2" to 2,
        "3" to 3,
        "4" to 4,
        "5" to 5
    )

    @FigmaProperty(FigmaType.Enum, "State")
    val readOnly: Boolean = Figma.mapping(
        "Default" to false,
        "Readonly" to true
    )

    @Composable
    fun ComponentExample() {
        KozmosRating(
            value = value,
            onValueChange = {},
            max = 5,
            readOnly = readOnly
        )
    }
}

File: packages/ios/Sources/Components/Rating/Rating.figma.swift
import SwiftUI
import Figma

struct KozmosRatingConnect: FigmaConnect {
    let component = KozmosRating.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=512-39747"

    @FigmaEnum(
        "Value",
        mapping: [
            "0": 0,
            "1": 1,
            "2": 2,
            "3": 3,
            "4": 4,
            "5": 5
        ]
    )
    var value: Int = 3

    @FigmaEnum(
        "State",
        mapping: [
            "Default": false,
            "Readonly": true
        ]
    )
    var readOnly: Bool = false

    var body: some View {
        KozmosRating(
            value: .constant(self.value),
            max: 5,
            readOnly: self.readOnly
        )
    }
}
